receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go

```diff
@@ -113,68 +113,45 @@ func (s *scraper) scrapeAndAppendNetworkCounterMetrics(metrics pdata.MetricSlice
 	if len(ioCounters) > 0 {
 		startIdx := metrics.Len()
 		metrics.EnsureCapacity(startIdx + networkMetricsLen)
-		initializeNetworkPacketsMetric(metrics.AppendEmpty(), metadata.Metrics.SystemNetworkPackets, startTime, now, ioCounters)
-		initializeNetworkDroppedPacketsMetric(metrics.AppendEmpty(), metadata.Metrics.SystemNetworkDropped, startTime, now, ioCounters)
-		initializeNetworkErrorsMetric(metrics.AppendEmpty(), metadata.Metrics.SystemNetworkErrors, startTime, now, ioCounters)
-		initializeNetworkIOMetric(metrics.AppendEmpty(), metadata.Metrics.SystemNetworkIo, startTime, now, ioCounters)
+
+		s.recordNetworkPacketsMetric(now, ioCounters)
+		s.recordNetworkDroppedPacketsMetric(now, ioCounters)
+		s.recordNetworkErrorPacketsMetric(now, ioCounters)
+		s.recordNetworkIOMetric(now, ioCounters)
+		s.mb.Emit(metrics)
```

Comment:
Just noticed that `Emit` called twice. It should be called only once per scrape. So we can rename `scrapeAndAppendNetworkCounterMetrics` to `recordNetworkCounterMetrics`, it should not accept any arguments and only call other `record...` functions. The same applies to `scrapeAndAppendNetworkConnectionsMetric`. Then `s.mb.Emit` should be called at the end of `scrape` function.

Reply:
Thanks for the comment. Fixed.